src/Histogram.hpp

```diff
@@ -43,4 +48,34 @@ class shasta::Histogram : public vector<size_t> {
 };
 
 
+// A histogram class that calculates bin size and finds the appropriate bin to increment given an x value
+class shasta::IterativeHistogram{
```

Comment:
Why is this called an `IterativeHistogram`? I see nothing iterative here.

Reply:
Perhaps just call it `Histogram2`, and at some point we can remove the more basic, existing `Histogram` class using `Histogram2` instead, then finally rename `Histogram2` to `Histogram`.

Reply:
Maybe "incremental" or "accumulative" would have made more sense, but Histogram2 is fine

Reply:
Yes, and if we do what I said above, it will eventually be renamed `Histogram` once the current `Histogram` class goes away.